hexrd/ui/llnl_import_tool_dialog.py

```diff
@@ -473,18 +476,25 @@ def completed(self):
                 [UI_TRANS_INDEX_ROTATE_90] * len(self.detectors))
         det_names = HexrdConfig().detector_names
         files = [[self.edited_images[det]['img']] for det in det_names]
-        ImageLoadManager().read_data(files, parent=self.ui)
+        ImageLoadManager().read_data(files, parent=self.ui.parent())
 
         buffer_default = {'status': 0}
         for det in det_names:
             det_config = HexrdConfig().config['instrument']['detectors'][det]
             buffer = det_config.setdefault('buffer', buffer_default)
             buffer['value'] = self.edited_images[det]['panel_buffer']
 
-        self.reset_panel()
+        self.close_widget(accepted=True)
         self.parent().action_show_toolbar.setEnabled(True)
         self.parent().action_show_toolbar.setChecked(True)
         self.cmap.block_updates(False)
 
     def show(self):
         self.ui.show()
+
+    def close_widget(self, accepted=False):
+        self.reset_panel()
+        if not accepted:
+            self.ui.cancel.clicked.connect(self.cancel_workflow.emit)
+        if self.ui.isFloating():
+            self.ui.close()
```

Comment:
I think `self.cancel_workflow.emit()` needs to be called, rather than a new connection being made, right?

And actually, I didn't realize you'd have to add an extra argument and extra logic to implement this correctly. Maybe we can instead change this function back to the way it was before, and add a function like this:
```python
def on_canceled(self):
    self.close_widget()
    self.cancel_workflow.emit()
```

The `self.ui.cancel.clicked` signal can then be connected to this instead.

Reply:
That connect should not have been there, that was a sloppy copy and paste, sorry. Cleaned up and updated to use the `on_canceled` method.

Reply:
Looks good now